src/Sarif.Converters/HdfConverter.cs

```diff
@@ -0,0 +1,194 @@
+        {
+            var reportingDescriptor = new ReportingDescriptor
+            {
+                Id = execJsonControl.Id,
+                Name = execJsonControl.Title,
+                ShortDescription = new MultiformatMessageString
+                {
+                    Text = AppendPeriod(execJsonControl.Desc),
+                },
+                DefaultConfiguration = new ReportingConfiguration
+                {
+                    Level = SarifLevelFromHdfImpact(execJsonControl.Impact),
+                },
+                HelpUri = null,
+                Relationships = new List<ReportingDescriptorRelationship>(
+                    ((JArray)execJsonControl.Tags["nist"])
+                    .Select(p => p.ToString().Trim()).Where(p => !string.IsNullOrWhiteSpace(p)).ToList().OrderBy(o => o)
+                    .Select(s => new ReportingDescriptorRelationship()
+                    {
+                        Target = new ReportingDescriptorReference()
+                        {
+                            Id = s,
+                            ToolComponent = new ToolComponentReference()
+                            {
+                                Name = "NIST",
+                                Guid = "AAFBAB93-5201-419E-8443-D4925C542398"
+                            }
+                        },
+                        Kinds = new List<string>() { "relevant" },
+                    }))
+            };
+
+            var results = new List<Result>();
+            foreach (ControlResult controlResult in execJsonControl.Results)
+            {
+                var result = new Result
+                {
+                    RuleId = execJsonControl.Id,
+                    Message = new Message
+                    {
+                        Text = AppendPeriod(controlResult.CodeDesc),
+                    },
+                    Level = SarifLevelFromHdfImpact(execJsonControl.Impact),
+                    Rank = SarifRankFromHdfImpact(execJsonControl.Impact),
+                };
+                results.Add(result);
+            }
+            return (reportingDescriptor, results);
+        }
+
+        private static FailureLevel SarifLevelFromHdfImpact(double impact)
+        {
+            // level 4 & 5
+            // Hdf: Critical, High
+            if (impact >= 0.7)
+            {
+                return FailureLevel.Error;
+            }
+            // level 3
+            // Hdf: Medium
+            else if (impact >= 0.5)
+            {
+                return FailureLevel.Warning;
+            }
+            // level 2
+            // Hdf: Low
+            else if (impact >= 0.3)
+            {
+                return FailureLevel.Warning;
+            }
+            // level 0, 1
+            // Hdf: Best_Practice, Information
+            else
+            {
+                return FailureLevel.Note;
+            }
+        }
+
+        private static double SarifRankFromHdfImpact(double impact) =>
```

Comment:
>private [](http://example.com/codeflow?start=8&length=7)

why/when do u choose to use private/internal?

Reply:
I copied the structure from existing converter to keep it similar as possible, so I did not gave it a thought. :)

I think generally, if we know this is currently called, or in future will be called out side of class we can make it internal.
we can also minimize scope as private and open as needed.

Reply:
mark everything as private if needed. if you will need to test something specific and direct, mark internal, so it can be accessed in the test projects